src/js/components/Header/Header.js

```diff
@@ -26,21 +26,19 @@ const Header = React.forwardRef(({ sticky, background, ...rest }, ref) => {
         runEventListener = false;
         setStickyStyles(undefined);
```

Comment:
When I first scroll down, the header abruptly disappears (position is not fixed yet then changes to fixed). I think this is caused by a similar issue to the other comment I had made previously that you resolved by always having position fixed and transition when the user has scrolled down (whether they are scrolling up or down).

That being said is there any reason we are not just always having the transition, position fixed, etc. when a sticky header is desired? Feels like we need that to be persistent in order for the animation to work and I'm not seeing the benefit of removing the sticky styles when we are scroll position 0.

Notice how the header abruptly disappears on my first scroll down but after that the transition works as expected (when I'm not at scrollY = 0), but then when I get back to scrollY = 0 at the end the header abruptly comes in.

https://user-images.githubusercontent.com/12522275/161646529-de011523-12de-4b17-832f-04b2daee677e.mov

Reply:
I agree I changed the logic around this I think the transition is better now. @ericsoderberghp I would like to hear your thoughts around this direction as well. 

I found that this is not working when in a `Layer` because of the `window.addEventListener` is not being triggered since it is a modal trying to solve for this now if we are good with the transition on the `Header`

Reply:
Agreed that the transitions shouldn't be removed without having fully run.

One potential reason to not use position fixed all of the time is that could prevent the Header's size from being influenced by its container. For instance, if, for whatever reason, the caller turns off `flex={false}` for the Header, such that a Box that contains it determines it's height. Or, perhaps the caller puts the Header inside a Grid area (we should test this scenario, since the added Box might not have the `gridArea` set) and the height is controlled by the Grid row.

Would it work to not add the sticky version until scrolling up and then delay removing the sticky version until after the transition finishes when scrolling down or reaching the top?

Reply:
that’s what I’d recommend trying